proxmox/nodes/vms/custom_storage_device.go

```diff
@@ -379,7 +376,8 @@ func (d CustomStorageDevices) Filter(fn func(*CustomStorageDevice) bool) CustomS
 // EncodeValues converts a CustomStorageDevices array to multiple URL values.
 func (d CustomStorageDevices) EncodeValues(_ string, v *url.Values) error {
 	for s, d := range d {
-		if d.Enabled {
+		// Explicitly skip disks which have FileID set, so it won't be encoded in "Create" or "Update" operations.
+		if d.FileID == nil || *d.FileID == "" {
```

Comment:
This logic is really hard to reason about. I think we should refactor this use case (as I recall, it is about creating a custom disk from a template) and use different data type there.